lib/pbench/server/api/resources/users_api.py

```diff
@@ -284,24 +285,55 @@ def post(self):
                     }
         """
         auth_token = Auth.get_auth_token()
-        user = Auth.verify_auth(auth_token=auth_token)
+        state = Auth.verify_internal_token(auth_token)
 
-        # "None" user represents that either the token is not present in our database or it is an expired token.
-        # Expired token is already deleted by now if we reach here.
-        # In either case we will return suceess to a client, since the same token can not be used again.
+        # The only time a token won't be in the database is if the token is
+        # not a valid token.  Valid tokens that are not expired should be in
+        # the database, and expired tokens might or might not be in the
+        # database.  If the valid token is found in the database associated
+        # with the user, remove it.
 
-        if user:
-            try:
-                ActiveTokens.delete(auth_token)
-            except Exception:
-                current_app.logger.exception(
-                    "Exception occurred deleting an auth token"
-                )
-                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-            else:
-                current_app.logger.debug("User {} logged out", user.username)
+        if state == Auth.TokenState.INVALID:
+            current_app.logger.info("User logout with invalid token: {}", auth_token)
         else:
-            current_app.logger.info("User logout with invalid or expired token")
+            token = ActiveTokens.query(auth_token)
+            if token:
+                user = token.user
+                if not user:
+                    # This is an internal error state, but play it cool with the
+                    # API client.
+                    current_app.logger.error(
+                        "INTERNAL ERROR - token {!r} ({}) found without an associated user",
+                        auth_token,
+                        state,
+                    )
```

Comment:
The token is (apparently) valid, and we found it in the database; why do we avoid removing it?  (If it had had a `user`, we _would_ have proceeded with the removal....)

Reply:
Actually, the database relationships and constraints should ensure that every token has a user.  So if we have a token without a user that would be an internal error related to a database problem of some sort.

I'll change this code to take advantage of that assertion.